pkg/ucp/server/server.go

```diff
@@ -17,192 +17,45 @@ limitations under the License.
-	if err != nil {
-		return Options{}, err
-	}
-
-	storeOpts := opts.Config.DatabaseProvider
-	planes := opts.Config.Planes
-	secretOpts := opts.Config.SecretProvider
-	qproviderOpts := opts.Config.QueueProvider
-	metricsOpts := opts.Config.MetricsProvider
-	traceOpts := opts.Config.TracerProvider
-	profilerOpts := opts.Config.ProfilerProvider
-	loggingOpts := opts.Config.Logging
-	identity := opts.Config.Identity
-	// Set the default authentication method if AuthMethod is not set.
-	if identity.AuthMethod == "" {
-		identity.AuthMethod = hostoptions.AuthDefault
-	}
-
-	location := opts.Config.Location
-	if location == "" {
-		location = "global"
-	}
-
-	var cfg *kube_rest.Config
-	if opts.Config.UCP.Kind == config.UCPConnectionKindKubernetes {
-		cfg, err = kubeutil.NewClientConfig(&kubeutil.ConfigOptions{
-			// TODO: Allow to use custom context via configuration. - https://github.com/radius-project/radius/issues/5433
-			ContextName: "",
-			QPS:         kubeutil.DefaultServerQPS,
-			Burst:       kubeutil.DefaultServerBurst,
-		})
-		if err != nil {
-			return Options{}, fmt.Errorf("failed to get kubernetes config: %w", err)
-		}
-	}
-
-	ucpConn, err := config.NewConnectionFromUCPConfig(&opts.Config.UCP, cfg)
-	if err != nil {
-		return Options{}, err
-	}
-
-	return Options{
-		Config:                  opts.Config,
-		Port:                    port,
-		TLSCertDir:              tlsCertDir,
-		PathBase:                basePath,
-		DatabaseProviderOptions: storeOpts,
-		SecretProviderOptions:   secretOpts,
-		QueueProviderOptions:    qproviderOpts,
-		MetricsProviderOptions:  metricsOpts,
-		TracerProviderOptions:   traceOpts,
-		ProfilerProviderOptions: profilerOpts,
-		LoggingOptions:          loggingOpts,
-		InitialPlanes:           planes,
-		Identity:                identity,
-		UCPConnection:           ucpConn,
-		Location:                location,
-	}, nil
-}
-
 // NewServer creates a new hosting.Host instance with services for API, EmbeddedETCD, Metrics, Profiler and Backend (if
 // enabled) based on the given Options.
-func NewServer(options *Options) (*hosting.Host, error) {
+func NewServer(options *ucp.Options) (*hosting.Host, error) {
 	hostingServices := []hosting.Service{
-		api.NewService(api.ServiceOptions{
-			ProviderName:            UCPProviderName,
-			Address:                 ":" + options.Port,
-			PathBase:                options.PathBase,
-			Config:                  options.Config,
-			Location:                options.Location,
-			TLSCertDir:              options.TLSCertDir,
-			DatabaseProviderOptions: options.DatabaseProviderOptions,
-			SecretProviderOptions:   options.SecretProviderOptions,
-			QueueProviderOptions:    options.QueueProviderOptions,
-			InitialPlanes:           options.InitialPlanes,
-			Identity:                options.Identity,
-			UCPConnection:           options.UCPConnection,
-		}),
+		api.NewService(options),
+		backend.NewService(options),
```

Comment:
This is the main simplification needed to be testable. The other changes in this PR serve to enable this simplicity.